File: Matlab/calculus/find_root/use/use_func_fixed_point.m
% Start
clc;
clear;
close all;

% Function
func = @(x)(x - 1 - 0.3.*cos(x));
Gfunc = @(x)(1 + 0.3.*cos(x));

% Input
ax = 1.5;
err = 0.00001;
Niter = 100;

% Algorithm
fprintf('I\ta\t\t\t\tg(a)\t\t\tf(a)\n');
for iter = 1:Niter
    NsolF = Gfunc(ax);
    NvalF = func(NsolF);
    fprintf('%d)\t%.10f\t%.10f\t%.10f\n', iter, ax, NsolF, NvalF);
    if abs(NvalF) < err
        break;
    end
    ax = NsolF;
end

% Output
fprintf('\nNsolF = %.10f\n', NsolF);

% Compare
EsolF = fzero(func, ax);
fprintf('EsolF = %.10f\nError = %.10f\n', EsolF, abs(EsolF - NsolF));

% **************************************************^**************************************************
% *****************************# Copyright by Ali Forouzandeh Hafshejani #*****************************
% ******************************# Scientific Computing Specialist 20@21 #******************************
% **************************************************^**************************************************

File: R/calculus/find_root/use/use_func_fixed_point.r
# Start

# Function
func = function(x)
{
    return (
        x - 1 - 0.3*cos(x)
    );
}

Gfunc = function(x)
{
    return (
        1 + 0.3*cos(x)
    );
}

# Input
ax = 1.5;
err = 0.00001;
Niter = 100;

# Algorithm
cat(sprintf("I\ta\t\tg(a)\t\tf(a)\n"));

for (iter in 1:Niter)
{
    NsolF = Gfunc(ax);
    NvalF = func(NsolF);

    cat(sprintf("%d)\t%.10f\t%.10f\t%.10f\n",
        iter, ax, NsolF, NvalF));

    if (abs(NvalF) < err)
    {
        break;
    }
    ax = NsolF;
}

# Output
cat(sprintf("\nNsolF = %.10f\n", NsolF));

# **************************************************^**************************************************
# *****************************# Copyright by Ali Forouzandeh Hafshejani #*****************************
# ******************************# Scientific Computing Specialist 20@21 #******************************
# **************************************************^**************************************************